shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1233,7 +1311,20 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
           case _ => mthCallOrSel(obj, fieldName)
         }
       case Let(isrec, nme, rhs, bod) =>
-        if (newDefs && !isrec) {
+        if (ctx.inQuote) {
+          val rhs_ty = typeTerm(rhs)
+          ctx.enterQuotedScope.poly {
+            newCtx => {
+              newCtx += nme.name -> VarSymbol(rhs_ty, nme)
+              val res_ty = typeTerm(bod)(newCtx, raise, vars, genLambdas)
+              val ctxTy = freshVar(noTyProv, N)(ctx.lvl)
+              con(newCtx.getCtxTy, newCtx.quoteSkolemEnv.foldLeft[ST](ctxTy)((res, ty) => ty._2 | res), TopType)(ctx)
+              ctx.trackFVs(ctxTy)
```

Comment:
Same suggestion as above, but instead of duplicating this code, it would make more sense to move it to some helper function in `Ctx`.

Reply:
Since this part invokes `con` function and depends on two contexts, I only refactor the `foldLeft` expression via a helper function in `Ctx`.